mlabs/src/Mlabs/NFT/Governance/Validation.hs

```diff
@@ -36,19 +77,282 @@ instance ValidatorTypes GovManage where
 
 -- | Minting policy for GOV and xGOV tokens.
 mkGovMintPolicy :: NftAppInstance -> GovAct -> ScriptContext -> Bool
```

Comment:
Add a general test that verifies that the Minting Policy and the `ListGovToken` in the `Head` have the same `CurrencySymbol`. This way we know that our  Minting Policy only mints for the correct List/Script.

Reply:
In every action case we look for the head datum in the inputs.
The function that returns head datum is `inputsWithHeadDatum`. 
```
    inputsWithHeadDatum :: [(GovDatum, TxOut)]
    inputsWithHeadDatum =
      filter
        ( \(datum', txO) ->
            datumIsHead (gov'list datum') && tokenPresent txO
        )
        $ getInputDatumsWithTx @GovDatum ctx
```
If you check it out, one of the checks that it performs is `tokenPresent` which itself checks the existence of our proofToken 
```
    symbol = ownCurrencySymbol ctx
    asset tn = AssetClass (symbol, TokenName tn)
    proofToken = asset emptyByteString
    
    tokenPresent txO =
      assetClassValueOf (txOutValue txO) proofToken == 1
        && assetClassValueOf (txOutValue txO) uniqueToken >= 1
```
Therefore it's already checked that the head datum contains the `listGov` token with our own `CurrencySymbol`